ethereum/kycdao-ntnft/test/KycdaoNTNFT.ts

```diff
@@ -168,40 +273,183 @@ describe.only('KycdaoNtnft Membership', function () {
+        await expect(receiver.mint(memberNft.address, 123, {value: expectedMintCost.mul(10)})).to.be.revertedWith('Refund failed')
+        expect(await memberNft.balanceOf(receiver.address)).to.equal(0)
+      })
+
+      it('Fails if receiver is a contract which attempts to reenter mint on onERC721Received', async function () {
+        const ReenterTestWalletAbstract = await ethers.getContractFactory('ReenterTestWallet')
+        const receiver = await ReenterTestWalletAbstract.deploy()
+        await memberNftAsMinter.authorizeMinting(123, receiver.address, testMetaUID, testVerifUID, expiration, true)
+        await expect(receiver.mint(memberNft.address, 123)).to.be.revertedWith('Unauthorized code')
+        expect(await memberNft.balanceOf(receiver.address)).to.equal(0)
+      })
+      
+      it('Fails if receiver is a contract which attempts to reenter on authorizeMinting', async function () {
+        const sendGasOnAuth = 100000
+        await memberNft.setSendGasOnAuthorization(sendGasOnAuth)
+        const ReenterAuthorizeTestWalletAbstract = await ethers.getContractFactory('ReenterAuthorizeTestWallet')
+        const receiver = await ReenterAuthorizeTestWalletAbstract.deploy()
+        await expect(memberNftAsMinter.authorizeMinting(123, receiver.address, testMetaUID, testVerifUID, expiration, false)).to.be.revertedWith('Failed to send gas')
+        await expect(receiver.mint(memberNft.address, 123, {value: expectedMintCost})).to.be.revertedWith('Unauthorized code')
+        expect(await memberNft.balanceOf(receiver.address)).to.equal(0)
+      })
+
+      it('Succeeds if receiver is a contract and send gas on auth is enabled', async function () {
+        const sendGasOnAuth = 100000
+        await memberNft.setSendGasOnAuthorization(sendGasOnAuth)
+        await minter.sendTransaction({ to: memberNft.address, value: sendGasOnAuth })
+        const EventReceiveTestWallettAbstract = await ethers.getContractFactory('EventReceiveTestWallet')
+        const receiver = await EventReceiveTestWallettAbstract.deploy()
+        await expect(memberNftAsMinter.authorizeMinting(123, receiver.address, testMetaUID, testVerifUID, expiration, false)).to.emit(receiver, 'ReceivedRefund')
+        expect(await ethers.provider.getBalance(receiver.address)).to.equal(sendGasOnAuth)
+        await receiver.mint(memberNft.address, 123, {value: expectedMintCost})
+        expect(await memberNft.balanceOf(receiver.address)).to.equal(1)
+      })
+    })
   })
 
-  describe('status', function () {
+  describe('getting metadata from KYCNFTs', function () {
+    describe('metadata', function () {
+      it('returns the expected tokenURI', async function () {
+        await memberNftAsMinter.authorizeMinting(123, anyone.address, testMetaUID, testVerifUID, expiration, false)
+        await memberNftAsAnyone.mint(123, {value: expectedMintCost})
+        expect(await memberNft.tokenURI(1)).to.equal(testInitArgs.baseURI + testMetaUID)
```

Comment:
this should probably use memberNftAsAnyone to show that it's publicly available

Reply:
Yea good point! Will change them now. Thanks!